### PR TITLE
API-40037-add-rescue-PoaVBMSUploadJob

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Adds generic rescue to PoaVBMSUploadJob
- adds test

## Related issue(s)

- [API-40052](https://jira.devops.va.gov/browse/API-40052)

## Testing done

- [X] *New code is covered by unit tests*
- force an error for a V1 POA upload {{uri}}/services/claims/v1/forms/2122/9a02384f-0104-4892-a9f2-85ac62a51515
- raise ::Common::Exceptions::BackendServiceException.new [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/sidekiq/claims_api/poa_vbms_upload_job.rb#L27)
```
        error_original_body = {
          messages: [
            {
              'key' => 'form526.submit.establishClaim.serviceError',
              'severity' => 'FATAL',
              'text' => 'Claim not established. System error with BGS. GUID: 00797c5d-89d4-4da6-aab7-24b4ad0e4a4f'
            }
          ]
        }
        raise Common::Exceptions::BackendServiceException.new(
          backtrace.backtrace,
          {},
          400,
          error_original_body
        )
```

*** don't forget new.perform [here](https://github.com/department-of-veterans-affairs/vets-api/blob/master/modules/claims_api/app/controllers/claims_api/v1/forms/power_of_attorney_controller.rb#L83)

## What areas of the site does it impact?
	modified:   modules/claims_api/app/sidekiq/claims_api/poa_vbms_upload_job.rb
	modified:   modules/claims_api/spec/sidekiq/poa_vbms_upload_job_spec.rb
	new file:   spec/support/vcr_cassettes/claims_api/bd/upload_error.yml

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/app/sidekiq/claims_api/service_base.rb [hunk 1]
```diff
@@ -191,5 +191,12 @@ def veteran_file_number(auto_claim)
     def evss_service
       ClaimsApi::EVSSService::Base.new
     end
+
+    def rescue_generic_errors(power_of_attorney, e)
+      power_of_attorney.status = ClaimsApi::PowerOfAttorney::ERRORED
+      power_of_attorney.vbms_error_message = e&.message || e&.original_body
+      power_of_attorney.save
+      ClaimsApi::Logger.log('PoaVBMSUploadJob', message: "In generic rescue, the error is: #{e}")
```
Comment (tycol7) [inline modules/claims_api/app/sidekiq/claims_api/service_base.rb:199]: Here and in https://github.com/department-of-veterans-affairs/vets-api/pull/18428: should we change `'PoaVBMSUploadJob'` to be something generic?

File: spec/support/vcr_cassettes/claims_api/bd/upload_error.yml
Comment (tycol7) [inline spec/support/vcr_cassettes/claims_api/bd/upload_error.yml:1]: Interesting that the status from BD is 400 but the body says success: true 😅

**Review by tycol7 — DISMISSED** [review]

File: modules/claims_api/app/sidekiq/claims_api/service_base.rb (hunk 1 above)
Comment (stiehlrod) [reply modules/claims_api/app/sidekiq/claims_api/service_base.rb]: @tycol7 Good call! Do mind re-stamping?

File: spec/support/vcr_cassettes/claims_api/bd/upload_error.yml
Comment (stiehlrod) [reply spec/support/vcr_cassettes/claims_api/bd/upload_error.yml:1]: Yeah it should say accepted since it has a sidekiq job

**Review by tycol7 — APPROVED** [review]

**Pull request merged**